Code/GraphMol/WedgeBonds.cpp

```diff
@@ -0,0 +1,494 @@
+  }
+
+  INT_VECT nChiralNbrs(mol.getNumAtoms(), noNbrs);
+
+  // start by looking for bonds that are already wedged
+  for (const auto bond : mol.bonds()) {
+    if (bond->getBondDir() == Bond::BEGINWEDGE ||
+        bond->getBondDir() == Bond::BEGINDASH ||
+        bond->getBondDir() == Bond::UNKNOWN) {
+      if (bond->getBeginAtom()->getChiralTag() == Atom::CHI_TETRAHEDRAL_CW ||
+          bond->getBeginAtom()->getChiralTag() == Atom::CHI_TETRAHEDRAL_CCW) {
+        nChiralNbrs[bond->getBeginAtomIdx()] = noNbrs + 1;
+      } else if (bond->getEndAtom()->getChiralTag() ==
+                     Atom::CHI_TETRAHEDRAL_CW ||
+                 bond->getEndAtom()->getChiralTag() ==
+                     Atom::CHI_TETRAHEDRAL_CCW) {
+        nChiralNbrs[bond->getEndAtomIdx()] = noNbrs + 1;
+      }
+    }
+  }
+
+  // now rank atoms by the number of chiral neighbors or Hs they have:
+  bool chiNbrs = false;
+  for (const auto at : mol.atoms()) {
+    if (nChiralNbrs[at->getIdx()] > noNbrs) {
+      // std::cerr << " SKIPPING1: " << at->getIdx() << std::endl;
+      continue;
+    }
+    auto type = at->getChiralTag();
+    if (type != Atom::CHI_TETRAHEDRAL_CW && type != Atom::CHI_TETRAHEDRAL_CCW) {
+      continue;
+    }
+    nChiralNbrs[at->getIdx()] = 0;
+    chiNbrs = true;
+    for (const auto nat : mol.atomNeighbors(at)) {
+      if (nat->getAtomicNum() == 1) {
+        // special case: it's an H... we weight these especially high:
+        nChiralNbrs[at->getIdx()] -= 10;
+        continue;
+      }
+      type = nat->getChiralTag();
+      if (type != Atom::CHI_TETRAHEDRAL_CW &&
+          type != Atom::CHI_TETRAHEDRAL_CCW) {
+        continue;
+      }
+      nChiralNbrs[at->getIdx()] -= 1;
+    }
+  }
+  return std::make_pair(chiNbrs, nChiralNbrs);
+}
+
+//
+// Determine bond wedge state
+///
+Bond::BondDir determineBondWedgeState(const Bond *bond,
+                                      unsigned int fromAtomIdx,
+                                      const Conformer *conf) {
+  PRECONDITION(bond, "no bond");
+  PRECONDITION(bond->getBondType() == Bond::SINGLE,
+               "bad bond order for wedging");
+  const auto mol = &(bond->getOwningMol());
+  PRECONDITION(mol, "no mol");
+
+  auto res = bond->getBondDir();
+  if (!conf) {
+    return res;
+  }
+
+  Atom *atom;
+  Atom *bondAtom;
+  if (bond->getBeginAtom()->getIdx() == fromAtomIdx) {
+    atom = bond->getBeginAtom();
+    bondAtom = bond->getEndAtom();
+  } else {
+    atom = bond->getEndAtom();
+    bondAtom = bond->getBeginAtom();
+  }
+
+  auto chiralType = atom->getChiralTag();
+  TEST_ASSERT(chiralType == Atom::CHI_TETRAHEDRAL_CW ||
+              chiralType == Atom::CHI_TETRAHEDRAL_CCW);
+
+  // if we got this far, we really need to think about it:
+  std::list<int> neighborBondIndices;
+  std::list<double> neighborBondAngles;
+  auto centerLoc = conf->getAtomPos(atom->getIdx());
+  auto tmpPt = conf->getAtomPos(bondAtom->getIdx());
+  centerLoc.z = 0.0;
+  tmpPt.z = 0.0;
```

Comment:
It looks like we are assuming 2d conformations but not checking, is this done somewhere?

Reply:
There’s no such assumption being made and this is not new behavior. The code has always behaved this way: if you ask for wedged bonds in a 3D conformer it will do what you ask and give you wedging that works for the 2D projection of the 3D coordinates